src/java.base/share/classes/sun/util/locale/LanguageTag.java

```diff
@@ -462,48 +432,51 @@ public static String caseFoldTag(String tag) {
     }
 
     public static LanguageTag parseLocale(BaseLocale baseLocale, LocaleExtensions localeExtensions) {
-        LanguageTag tag = new LanguageTag();
 
-        String language = baseLocale.getLanguage();
-        String script = baseLocale.getScript();
-        String region = baseLocale.getRegion();
-        String variant = baseLocale.getVariant();
+        String language = EMPTY_TAG;
+        String script = EMPTY_TAG;
+        String region = EMPTY_TAG;
+
+        String baseLanguage = baseLocale.getLanguage();
+        String baseScript = baseLocale.getScript();
+        String baseRegion = baseLocale.getRegion();
+        String baseVariant = baseLocale.getVariant();
 
         boolean hasSubtag = false;
 
         String privuseVar = null;   // store ill-formed variant subtags
 
-        if (isLanguage(language)) {
+        if (isLanguage(baseLanguage)) {
             // Convert a deprecated language code to its new code
-            if (language.equals("iw")) {
-                language = "he";
-            } else if (language.equals("ji")) {
-                language = "yi";
-            } else if (language.equals("in")) {
-                language = "id";
+            if (baseLanguage.equals("iw")) {
+                baseLanguage = "he";
+            } else if (baseLanguage.equals("ji")) {
+                baseLanguage = "yi";
+            } else if (baseLanguage.equals("in")) {
+                baseLanguage = "id";
             }
-            tag.language = language;
+            language = baseLanguage;
```

Comment:
```suggestion
            language = switch(baseLanguage) {
                case "iw" -> "he";
                case "ji" -> "yi";
                case "in" -> "id";
                default   -> baseLanguage;
            };
```
use switch expression